rubicon_ml/client/rubicon_json.py

```diff
@@ -1,14 +1,107 @@
 from jsonpath_ng.ext import parse
 
-from rubicon_ml.client import Experiment, Project, Rubicon
+from rubicon_ml.client import (
+    Artifact,
+    Dataframe,
+    Experiment,
+    Feature,
+    Metric,
+    Parameter,
+    Project,
+    Rubicon,
+)
+from rubicon_ml.domain import (
+    Artifact as DomainArtifact,
+    Dataframe as DomainDataframe,
+    Experiment as DomainExperiment,
+    Feature as DomainFeature,
+    Metric as DomainMetric,
+    Parameter as DomainParameter,
+    Project as DomainProject,
+)
+
+
+class NoOpParent:
+    """A read-only parent object"""
+
+    @property
+    def _config(self):
+        return None
 
 
 class RubiconJSON:
     def __init__(self, rubicon_objects=None, projects=None, experiments=None):
         self._json = self._convert_to_json(rubicon_objects, projects, experiments)
 
-    def search(self, query):
-        return parse(query).find(self._json)
+    def search(self, query, return_type=None):
+        if return_type is not None:
+            if return_type.lower() not in [
+                "artifact",
+                "dataframe",
+                "experiment",
+                "feature",
+                "metric",
+                "parameter",
+                "project",
+            ]:
+                raise ValueError(
+                    "`return_type` must be artifact, dataframe, experiment, featrure, metric, parameter, or project."
+                )
+
+        res = parse(query).find(self._json)
+        if not return_type:
+            return res
+
+        return_objects = []
+        if return_type == "artifact":
+            for match in res:
+                for i in range(len(match.value)):
+                    return_objects.append(Artifact(DomainArtifact(**match.value[i]), NoOpParent()))
+        elif return_type == "dataframe":
+            for match in res:
+                for i in range(len(match.value)):
+                    return_objects.append(
+                        Dataframe(DomainDataframe(**match.value[i]), NoOpParent())
+                    )
+        elif return_type == "experiment":
```

Comment:
I noticed this and "project" don't have the `for i in range(len(match.value)):` loop as well - is this intentional?

Reply:
Hey! Yeah so I intentionally removed the `for i in range(len(match.value)):` in the "experiment" and "project" conditions because I wasn't able to create a test case where `match.value` was a `list` for those two return_types. Furthermore, `match.value` was always being returned as a `list` for "artifact", "dataframe", "metric", and "parameter" and I couldn't create a test condition that would enter the `else` case where it wasn't a `list`.